e2e_tests/src/e2e_conduit/fixtures/plugin_fixture.py

```diff
@@ -0,0 +1,17 @@
+class PluginFixture:
+    def __init__(self):
+        self.config_input = {}
+        self.config_output = {}
+
+    @property
+    def name(self):
```

Comment:
I believe that `NotImplemented` is meant for usage when a special dunder method is not implemented, while `NotImplementedError` is meant for abstract methods that should be implemented by concrete subclasses. Some links:
* https://docs.python.org/3/library/constants.html#NotImplemented
* https://realpython.com/python-interface/#using-abstract-method-declaration